File: src/design/data/DesignGenerationRequest.ts
import type {DesignConceptData} from './DesignConceptData.js'
import type {DesignPreferenceProfileData} from './DesignPreferenceProfileData.js'
import type {DesignSourceMode} from './DesignIntentData.js'
export interface DesignGenerationRequest{readonly prompt:string;readonly sourceMode?:DesignSourceMode;readonly referenceImageUrl?:string;readonly targetUrl?:string;readonly selectedConcept?:DesignConceptData;readonly pages?:readonly string[];readonly preferenceProfile?:DesignPreferenceProfileData}

File: src/main/java/org/tavall/webdesign/design/data/DesignGenerationRequest.java
package org.tavall.webdesign.design.data;

import java.util.List;

public record DesignGenerationRequest(
        String prompt,
        DesignSourceMode sourceMode,
        String referenceImageUrl,
        String targetUrl,
        DesignConceptSet.DesignConcept selectedConcept,
        List<String> pages,
        DesignPreferenceProfile preferenceProfile
) {
    public DesignGenerationRequest {
        pages = pages == null ? List.of() : List.copyOf(pages);
    }
}
